src/hooks/useScanApi.ts

```diff
@@ -1,73 +1,448 @@
-/**
- * Scan API Hook
- *
- * This hook provides functions for interacting with scan-related API calls,
- * including running and deleting scans, managing report downloads.
- *
- * @module useScanApi
- */
-import React from 'react';
-import axios from 'axios';
-import { ScanType } from '../types/types';
-
-const useScanApi = () => {
-  const [pendingDeleteScan, setPendingDeleteScan] = React.useState<ScanType>();
-  const [scanSelected, setScanSelected] = React.useState<ScanType>();
-
-  /**
-   * Executes a DELETE request to delete a scan.
-   *
-   * @param {ScanType} [scan] - (Optional) The scan to be deleted. If not provided, it uses `pendingDeleteScan`.
-   * @returns {AxiosResponse} - The Axios response object representing the result of the request.
-   */
-  const deleteScan = (scan?: ScanType) => {
-    return axios.delete(`${process.env.REACT_APP_SCANS_SERVICE}${(scan || pendingDeleteScan)?.id}/`);
-  };
+import { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { type AlertProps } from '@patternfly/react-core';
+import axios, { AxiosError, type AxiosResponse, isAxiosError } from 'axios';
+import helpers from '../helpers';
+import apiHelpers from '../helpers/apiHelpers';
+import { Scan, ScanJobType, ScanJobsResponse, SourceType } from '../types/types';
```

Comment:
I have to evaluate the use of the types in our codebase. If I am not mistaken they are also used in our views. They MUST be used there? I am unsure about this, maybe we overcomplicate things there.

Reply:
We're probably going to need to clean types up anyway... it can be easier to expand a little when you're developing and then we go back and refine afterwards, in theory =)